tests/test_integration_color.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+from tests.utilities.test_app import AppTest
+from textual.app import ComposeResult
+from textual.color import Color, BLACK, WHITE
+from textual.geometry import Size
+from textual.widget import Widget
+
+
+@pytest.mark.parametrize(
+    "widget_color,widget_background_color,expected_color",
+    (
+        # "auto" on a white background --> foreground is black
+        ["auto", "white", BLACK],
+        # "auto" on a black background --> foreground is white
+        ["auto", "black", WHITE],
+        # And now with a pinch of alpha:
+        ["auto 50%", "white", WHITE.get_contrast_text(0.5)],
+        ["30% auto", "black", BLACK.get_contrast_text(0.3)],
+    ),
+)
+def test_color_auto_single_widget(
```

Comment:
here is a "closer to the metal" test for the "auto" colour, that can only test that a Widget adapts its "auto" colour to its own background :-)